Source/ARTPush.m

```diff
@@ -54,6 +55,16 @@ - (void)didFailToRegisterForRemoteNotificationsWithError:(NSError *)error {
     [_logger error:@"ARTPush: device token not received (%@)", [error localizedDescription]];
 }
 
++ (ARTPushActivationStateMachine *)activationMachine {
+    static dispatch_once_t once;
+    static id activationMachineInstance;
+    dispatch_once(&once, ^{
+        // Error: ARTPush.m:62:81: Instance variable '_httpExecutor' accessed in class method
+        //activationMachineInstance = [[ARTPushActivationStateMachine alloc] init:_httpExecutor];
+    });
```

Comment:
Yes, it should be a singleton. The persistent data it needs is the current state, the pending events, the device ID (which, once generated, doesn't change), the device's update token and the APNs device token. Also, whether activate/deactivate were called with custom registration or not.